test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorHostTest.cs

```diff
@@ -254,6 +286,23 @@ private static Stream GetResourceStream(string resourceName)
                 generatedLocation: new MappingLocation(generatedLocation, contentLength));
         }
 
+        private class PathValidatingChunkInheritanceUtility : ChunkInheritanceUtility
```

Comment:
Might be simpler to subclass `MvcRazorHost` in the tests and `new DecorateRazorParser` or just verify the data that you're looking for in the return values. Then you don't need any 'test only' code in the product.